src/MainWindow.cpp

```diff
@@ -1531,3 +1549,48 @@ QMessageBox::StandardButton MainWindow::openNewFileQuestionDialog(const QString&
           QMessageBox::Yes | QMessageBox::No, QMessageBox::No
     );
 }
+
+// TODO: Reuse code between generateBashRemediationRole and generateAnsibleRemediationRole
+void MainWindow::generateBashRemediationRole()
+{
```

Comment:
refactor this into a method that takes the short name "bash remediation role", extension ".sh" and OpenSCAP remediation template "urn:xccdf:...". Then the two generate functions can just call that. Shorter code is better.

Reply:
I took slightly different approach, so I hope you like it too as it has the same effect.